pkg/proxy/ipvs/proxier.go

```diff
@@ -696,6 +697,15 @@ func cleanupIptablesLeftovers(ctx context.Context, ipt utiliptables.Interface) (
 
 // CleanupLeftovers clean up all ipvs and iptables rules created by ipvs Proxier.
 func CleanupLeftovers(ctx context.Context) (encounteredError bool) {
+	// libipvs.New() will log errors if the "ip_vs" kernel module (or the "modprobe"
+	// binary) is not available. Logging an extra error is fine if we were actually
+	// trying to run the ipvs proxier, but it's confusing to see when just doing
+	// best-effort cleanup (eg, when starting the nftables proxier), so we do the same
+	// check libipvs does here, and bail out without calling libipvs if it fails.
+	if _, err := exec.Command("modprobe", "-va", "ip_vs").CombinedOutput(); err != nil {
```

Comment:
I was thinking we already had some discovery logic but I could not find it

Reply:
I think the ipvs proxy itself used to have some logic. But anyway, the point of this check is to fail in _exactly_ the same circumstances as libipvs would, so we don't care about other logic we might have anyway.